sql/src/main/java/io/crate/operation/scalar/ArrayDifferenceFunction.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar;
+
+import com.google.common.base.Preconditions;
+import io.crate.metadata.*;
+import io.crate.operation.Input;
+import io.crate.planner.symbol.Function;
+import io.crate.planner.symbol.Literal;
+import io.crate.planner.symbol.Symbol;
+import io.crate.types.ArrayType;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.common.Nullable;
+
+import java.util.*;
+
+public class ArrayDifferenceFunction extends Scalar<Object[], Object> {
+
+    public static final String NAME = "array_difference";
+    private FunctionInfo functionInfo;
+    private final Optional<Set<Object>> optionalSubtractSet;
```

Comment:
please don't use Optional from ``java.util`` as it is not available on java7 which we compile for.
Use ``com.google.common.base.Optional``